### PR TITLE
Fix issues in CFn resources and filter notfound exceptions for delete operations

### PR DESCRIPTION
**Pull request by dominikschubert**
## Motivation

Users reported simple updates with serverless/SAM/CDK failing, usually due to issues in individual resource providers, especially for delete operations caused by updates.

Fixes https://github.com/localstack/localstack/issues/9895
Fixes https://github.com/localstack/localstack/issues/10042
Fixes https://github.com/localstack/localstack/issues/10169

## Changes

- `"AWS::ApiGateway::Method"`: Fixed a few issues related to parameter transformation and passing
- `"AWS::ApiGateway::Deployment"`: Delete any stages that are still linked to this deployment before deleting the deployment itself
- `AWS::Lambda::Function`: Added a waiter after updating the function code. This was leading to intermittent 500 exceptions and retrying the handler. Didn't usually affect stack deployments but lead to some weird errors in the logs.
- Added filter for any "notfound" exceptions when trying to call the delete handler of a resource
- Added a regression test for this that simulates what the serverless deploy steps shown below would do

## Testing

```bash
# a serverless deploy is actually a CreateStack AND an UpdateStack operation
serverless deploy --stage=local
# change lambda handler name for example
serverless deploy --stage=local
```

### CODE REVIEW

File: tests/aws/services/cloudformation/resources/test_apigateway.validation.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
 {
+  "tests/aws/services/cloudformation/resources/test_apigateway.py::TestServerlessApigwLambda::test_serverless_like_deployment_with_update": {
+    "last_validated_date": "2024-02-19T08:55:12+00:00"
+  },
+  "tests/aws/services/cloudformation/resources/test_apigateway.py::TestServerlessApigwLambda::test_stuff": {
+    "last_validated_date": "2024-02-16T14:51:15+00:00"
+  },
+  "tests/aws/services/cloudformation/resources/test_apigateway.py::TestServerlessApigwLambda::test_update_trigger": {
+    "last_validated_date": "2024-02-16T22:21:12+00:00"
+  },
```
Comment (dominikschubert) [inline tests/aws/services/cloudformation/resources/test_apigateway.validation.json:10]:
```suggestion
```

File: localstack/services/apigateway/resource_providers/aws_apigateway_method.py [hunk 2]
```diff
@@ -104,7 +105,7 @@ def create(
         # key_to_lower makes in-place changes which will cause crash
         # when we try to use model outside of this function
         # for example generating composite physical id
-        params = keys_to_lower(model.copy())
+        params = keys_to_lower(deepcopy(model))
```
Comment (dominikschubert) [inline localstack/services/apigateway/resource_providers/aws_apigateway_method.py:108]: Long term we should fix this by using the provider utils and not keys_to_lower.

**Comment by dominikschubert** [comment]
Failures seem to be caused by the broken deletes due to us basically ignoring dependencies for deletes/replacements. Will need to look into it. We should probably try to ignore any deletes during stack updates for now :man_shrugging:

**Review by pinzon — APPROVED** [review]

**Comment by Yehonal** [comment]
Thank you! We can't wait to test it 
Do you know when the localstack-pro image will be updated on the Docker Hub?
I've seen the `localstack` one has been already updated

**Pull request merged**